R/validate.R

```diff
@@ -168,6 +168,92 @@
   return(NULL)
 }
 
+#' Validate the inputs to `estimate_and_apply_uncertainty()` to ensure that
+#'    the reporting triangle, point nowcast matrix, and specified maximum delay
+#'    are correct.
+#'
+#' @inheritParams estimate_and_apply_uncertainty
+#'
+#' @returns NULL, invisibly
+#' @keywords internal
+.validate_multiple_inputs <- function(
+    point_nowcast_matrix,
+    reporting_triangle,
+    max_delay) {
+  # Basic input validation
+  if (!is.matrix(point_nowcast_matrix)) {
+    cli_abort("`point_nowcast_matrix` must be a matrix.")
+  }
+
+  if (ncol(point_nowcast_matrix) != ncol(reporting_triangle)) {
+    cli_abort(c(
+      "x" = "`point_nowcast_matrix` and `reporting_triangle` must have the same number of columns.", # nolint
+      "i" = "Got {ncol(point_nowcast_matrix)} and {ncol(reporting_triangle)} respectively." # nolint
+    ))
+  }
+
+  if (ncol(reporting_triangle) != (max_delay + 1)) {
+    cli_abort(c(
+      "x" = "Inconsistent `max_delay`.", # nolint
+      "i" = "`ncol(reporting_triangle)` = {ncol(reporting_triangle)} but `max_delay + 1` = {max_delay + 1}." # nolint
+    ))
+  }
+
+
+
+
+  return(NULL)
+}
+
+#' Validate the specified number of reference times meets the minimum
+#'    requirements
+#'
+#' @param n_ref_times Integer indicating the number of reference times
+#'    available.
+#' @param size_min_ref_times_delay Integer indicating minimum number of
+#'    reference times needed for delay estimation.
+#' @inheritParams estimate_and_apply_uncertainty
+#' @inheritParams allocate_reference_times
+#' @autoglobal
+#' @returns NULL, invisibly
+.validate_ref_time_allocation <- function(n_ref_times,
```

Comment:
This looks like a duplication of other checks but I am quite lost in the different validation parts of the code.

Reply:
There are two validation functions that have similar names. 

This one checks that the `n_history_delay` and `n_retrospective_nowcasts` passed into the `estimate_and_apply_delay()` function are valid -- I have renamed to `validate_inputs_uncertainty()`

The other checks that the parameters to `allocate_reference_times are valid, I have nammed this to `.validate_inputs_allocation`

Reply:
and they share no code or overlap?